action/protocol/execution/evm/evm.go

```diff
@@ -189,7 +189,7 @@ func ExecuteContract(
 		blkCtx.BlockHeight,
 		hu.IsPre(config.Aleutian, blkCtx.BlockHeight),
 		hu.IsPost(config.Greenland, blkCtx.BlockHeight),
-		execution.Hash(),
+		actionCtx.ActionHash,
```

Comment:
this is a wrong programming: should use action hash in context

that is why SetEnvelopeContext() need to do
```
act.hash = Selp.Hash()
```